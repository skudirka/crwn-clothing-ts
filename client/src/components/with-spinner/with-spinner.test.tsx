import {shallow, mount, render} from 'enzyme';
import React from 'react';

import WithSpinner from './with-spinner.component';
import Spinner from '../spinner/spinner.component';

describe('WithSpinner HOC', () => {
    const TestComponent = () => <div className='inner' />;
    const WrappedComponent = WithSpinner(TestComponent);

    it('should render WithSpinner component and inner child', () => {
        const wrapper = shallow(<WrappedComponent isLoading={false} />);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.exists(Spinner)).toBe(false);
        expect(wrapper.exists(TestComponent)).toBe(true);
    });

    it('if loading, should render spinner and not the passed inner child', () => {
        const wrapper = shallow(<WrappedComponent isLoading={true} />);
        
        expect(wrapper.exists(Spinner)).toBe(true);
        expect(wrapper.exists(TestComponent)).toBe(false);
    });
    
})