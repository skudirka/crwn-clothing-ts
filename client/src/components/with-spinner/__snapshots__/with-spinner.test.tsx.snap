// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithSpinner HOC should render WithSpinner component and inner child 1`] = `ShallowWrapper {}`;
